### PR TITLE
Add ReportDetails component to the advisor-components library 

### PR DESCRIPTION
**Pull request by gkarat**
This PR adds another component to the advisor-components library: ReportDetails. The component is accountable for rendering rule results for a particular OpenShift cluster. As the next step, this should replace the current implementation at https://github.com/RedHatInsights/insights-advisor-frontend/tree/master/src/PresentationalComponents/ReportDetails.

- [x] Implement the component
- [x] Refactor code and stylesheets
- [x] Add sanitizer step after transforming Markdown files to HTML
- [x] Add basic tests and fixtures
- [x] Add docs

View in cypress:
Screenshot from 2022-05-25 14-50-56
Screenshot from 2022-05-25 14-51-28

View in RHEL Advisor (local deployment):
Screenshot 2022-05-25 at 15-10-49 msager-rhel8 - Systems - Advisor Red Hat Insights
Screenshot 2022-05-25 at 15-10-56 msager-rhel8 - Systems - Advisor Red Hat Insights

### CODE REVIEW

File: packages/advisor-components/src/ReportDetails/ReportDetails.spec.ct.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      />
+    );
+  });
+
+  it('renders component', () => {
+    cy.get(ROOT);
+  });
+
+  it('renders four headers', () => {
+    cy.get('.pf-c-card__header').should('have.length', 4);
```
Comment (ikerreyes) [inline packages/advisor-components/src/ReportDetails/ReportDetails.spec.ct.js:26]: Should have the length of the headers, not `4`.

**Review by ikerreyes — COMMENTED** [review]
I think we should rethink the data used in the test, see comments.

File: packages/advisor-components/cypress/fixtures/report.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{
+  "rule": {
+    "rule_id": "foobar_deprecated_something|FOOBAR_DEPRECATED_SOMETHING",
+    "node_id": "12345678",
+    "summary": "New Ansible Engine packages are inaccessible when dedicated Ansible repo is not enabled.\n",
```
[the remark this answers is not in the document]
Comment (gkarat) [reply packages/advisor-components/cypress/fixtures/report.json:5]: @ikerreyes, the content of the rules can be exposed; however, it is not generally suggested. Still, this fixture was taken from one of the API responses, but I have modified all of the identifying information here, so it shouldn't be a problem...

File: packages/advisor-components/src/ReportDetails/ReportDetails.spec.ct.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  });
+
+  it('renders code blocks with pre and code tags', () => {
+    cy.get('pre code').should('have.length', 7);
+  });
+
+  it('renders table correctly', () => {
+    cy.get('table > tbody').within(() => {
+      cy.get('tr').should('have.length', 2);
+      cy.get('tr > td').eq(0).should('contain', 'eth0');
```
[the remark this answers is not in the document]
Comment (gkarat) [reply packages/advisor-components/src/ReportDetails/ReportDetails.spec.ct.js:54]: yep, good point. I tried to replace the hardcoded value where it seemed to be straightforward. for other places I left todos :)

**Review by ikerreyes — APPROVED** [review]

**Review by Hyperkid123 — APPROVED** [review]

**Comment by Hyperkid123** [comment]
@gkarat I see you still have the docs TODO. Let me know if you want this to be merged and create docs later.

**Comment by gkarat** [comment]
@Hyperkid123, thank you, I added docs. if everything still looks fine, I am OK with merging it 👍🏼

**Pull request merged**